gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/resources/account/AccountEndpointHandler.java

```diff
@@ -0,0 +1,151 @@
+                                  ClientSyncService clientSyncService,
+                                  Domain domain) {
+        this.userService = userService;
+        this.factorService = factorService;
+        this.activityAuditService = activityAuditService;
+        this.clientSyncService = clientSyncService;
+        this.domain = domain;
+    }
+
+    public void getUser(RoutingContext routingContext) {
+        JWT token = routingContext.get(ConstantKeys.TOKEN_CONTEXT_KEY);
+        userService.findById(token.getSub()).doOnError(err -> {
+            //todo: handle error
+        }).toSingle().subscribe(user -> {
+            routingContext.put(ConstantKeys.USER_CONTEXT_KEY, user);
+            routingContext.next();
+        });
+    }
+
+    //mounted angular
+    public void getAccount(RoutingContext routingContext) {
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    //static assets
+    public void getAsset(RoutingContext routingContext) {
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    public void updateDevices(RoutingContext routingContext) {
+        //TODO: waiting to impl fully
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    public void getProfile(RoutingContext routingContext) {
+        AccountResponseHandler.handleDefaultResponse(routingContext, routingContext.get(ConstantKeys.USER_CONTEXT_KEY));
+    }
+
+    public void getActivity(RoutingContext routingContext) {
+        getActivityAudit(routingContext, routingContext.get(ConstantKeys.USER_CONTEXT_KEY)).subscribe(result -> AccountResponseHandler.handleDefaultResponse(routingContext, result));
+    }
+
+    public void getDevices(RoutingContext routingContext) {
+        collectFactors(routingContext.get(ConstantKeys.USER_CONTEXT_KEY)).subscribe(factors -> AccountResponseHandler.handleDefaultResponse(routingContext, factors));
+    }
+
+    public void updateProfile(RoutingContext routingContext) {
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        User updatedUser = routingContext.getBodyAsJson().mapTo(User.class);
+        if (Objects.equals(user.getId(), updatedUser.getId())) {
+            userService.update(updatedUser)
+                    .doOnSuccess(result -> AccountResponseHandler.handleUpdateUserResponse(routingContext))
+                    .doOnError(er -> AccountResponseHandler.handleUpdateUserResponse(routingContext, er.getMessage()));
+        } else {
+            AccountResponseHandler.handleUpdateUserResponse(routingContext, "Mismatched user IDs", 401);
+        }
+    }
+
+    private Single<Page<Audit>> getActivityAudit(RoutingContext routingContext, User user) {
+        int page;
```

Comment:
Having an utility method (or object) to build a "Page object" may be cleaner if we have to use pages in other endpoints.
Wdyt?

Reply:
we could migrate to a builder pattern for pages if that's something that you think would be clean, should we backlog it or start it on this commit? @leleueri

Reply:
In this commit, it will be fine if you have time.